File: src/org/redrune/networking/packet/context/impl/NPCInteractionPacketContext.java
package org.redrune.networking.packet.context.impl;

import org.redrune.game.GameFlags;
import org.redrune.game.content.entity.actor.player.event.npc.NPCInteractionEvent;
import org.redrune.game.entity.actor.npc.NPC;
import org.redrune.game.entity.actor.player.Player;
import org.redrune.networking.packet.context.PacketContext;
import org.redrune.utility.game.ClickOption;
import org.redrune.utility.game.repository.npc.characteristic.NPCCharacteristicRepository;
import org.redrune.utility.game.repository.npc.spawn.NPCSpawnRepository;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 2019-02-04
 */
public class NPCInteractionPacketContext extends PacketContext {
	
	private final NPC npc;
	
	private final ClickOption option;
	
	private final boolean running;
	
	public NPCInteractionPacketContext(NPC npc, ClickOption option, boolean running) {
		this.npc = npc;
		this.option = option;
		this.running = running;
	}
	
	@Override
	public void handle(Player player) {
		if (running) {
			player.setRunModeOn(true);
		}
		switch (option) {
			case FIRST:
			case SECOND:
			case THIRD:
			case FOURTH:
				player.getEventManager().start(new NPCInteractionEvent(npc, option));
				break;
		}
		
	}
}

File: src/main/kotlin/org/redrune/networking/packet/context/impl/NPCInteractionPacketContext.kt
package org.redrune.networking.packet.context.impl

import org.redrune.game.content.entity.actor.player.event.npc.NPCInteractionEvent
import org.redrune.game.entity.actor.npc.NPC
import org.redrune.game.entity.actor.player.Player
import org.redrune.networking.packet.context.PacketContext
import org.redrune.utility.game.ClickOption

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 2019-02-04
 */
class NPCInteractionPacketContext(private val npc: NPC, private val option: ClickOption, private val running: Boolean) :
    PacketContext() {
    override fun handle(player: Player) {
        if (running) {
            player.isRunModeOn = true
        }
        when (option) {
            ClickOption.FIRST, ClickOption.SECOND, ClickOption.THIRD, ClickOption.FOURTH -> player.eventManager.start(
                NPCInteractionEvent(
                    npc, option
                )
            )

            else -> {}
        }
    }
}